src/Magnum/DartIntegration/Test/DartSkeletonTest.cpp

```diff
@@ -0,0 +1,267 @@
+                                         .second;
+
+    // Make a shape for the Joint
+    const double R = default_width / 2.0;
+    const double h = default_depth;
+    std::shared_ptr<dart::dynamics::CylinderShape> cyl(new dart::dynamics::CylinderShape(R, h));
+
+    // Line up the cylinder with the Joint axis
+    Eigen::Isometry3d tf(Eigen::Isometry3d::Identity());
+    tf.linear() = dart::math::eulerXYZToMatrix(
+        Eigen::Vector3d(90.0 * M_PI / 180.0, 0, 0));
+
+    auto shapeNode = bn->createShapeNodeWith<dart::dynamics::VisualAspect>(cyl);
+    shapeNode->getVisualAspect()->setColor(dart::Color::Blue());
+    shapeNode->setRelativeTransform(tf);
+
+    // Set the geometry of the Body
+    setGeometry(bn);
+
+    return bn;
+}
+
+namespace Magnum { namespace DartIntegration { namespace Test {
+
+namespace {
+
+    using namespace Math::Literals;
+
+    typedef SceneGraph::Object<SceneGraph::MatrixTransformation3D> Object3D;
+    typedef SceneGraph::Scene<SceneGraph::MatrixTransformation3D> Scene3D;
+
+}
+
+struct DartSkeletonTest : TestSuite::Tester {
+    explicit DartSkeletonTest();
+
+    void test();
+    void test_urdf();
+    void test_texture();
+};
+
+DartSkeletonTest::DartSkeletonTest() {
+    addTests({&DartSkeletonTest::test});
+    addTests({&DartSkeletonTest::test_urdf});
+    addTests({&DartSkeletonTest::test_texture});
+}
+
+void DartSkeletonTest::test() {
+    // Create an empty Skeleton with the name "pendulum"
+    dart::dynamics::SkeletonPtr pendulum = dart::dynamics::Skeleton::create("pendulum");
+
+    // Add each body to the last BodyNode in the pendulum
+    dart::dynamics::BodyNode* bn = makeRootBody(pendulum, "body1");
+    bn = addBody(pendulum, bn, "body2");
+    bn = addBody(pendulum, bn, "body3");
+    bn = addBody(pendulum, bn, "body4");
+    bn = addBody(pendulum, bn, "body5");
+
+    // Create a world and add the pendulum to the world
+    dart::simulation::WorldPtr world(new dart::simulation::World);
+    world->addSkeleton(pendulum);
+
+    Scene3D scene;
+    Object3D* obj = new Object3D{&scene};
+
+    DartSkeleton* skel = new DartSkeleton{obj, pendulum};
+
+    // Set the initial joint positions so that the pendulum
+    // starts to swing right away
+    pendulum->getDof(1)->setPosition(120. * M_PI / 180.0);
+    pendulum->getDof(2)->setPosition(20. * M_PI / 180.0);
+    pendulum->getDof(3)->setPosition(-50. * M_PI / 180.0);
+
+    world->step();
+    skel->updateObjects();
+
+    auto objects = skel->objects();
+
+    auto trans = bn->getShapeNodesWith<dart::dynamics::VisualAspect>().back()->getTransform();
+
+    auto R = Eigen::AngleAxisd(trans.linear());
+    auto axis = R.axis();
+    auto T = trans.translation();
+
+    // Convert it to axis-angle representation
+    Math::Vector3<Float> t = {Float(T[0]), Float(T[1]), Float(T[2])};
+    Math::Vector3<Float> u = {Float(axis(0)), Float(axis(1)), Float(axis(2))};
+    Float theta = Float(R.angle());
```

Comment:
Similarly to my comment in `DartObject.cpp` above -- can the types be explicit instead of `auto`? Also, the math type conversion can be written in a shorter way.

> axis-angle representation

This caught my attention here, but applies to `DartObject,cpp` as well. Wouldn't there be some simpler, more direct way? (matrices? dual quats?) If I remember correctly, converting to axis/angle involves some nasty square roots and acos and so it's not exactly fast. I saw Eigen for the first time today, so I have no idea in what representation the `Isometry3d` type stores the data :D

Reply:
> This caught my attention here, but applies to DartObject,cpp as well. Wouldn't there be some simpler, more direct way? (matrices? dual quats?) If I remember correctly, converting to axis/angle involves some nasty square roots and acos and so it's not exactly fast. I saw Eigen for the first time today, so I have no idea in what representation the Isometry3d type stores the data :D

I did like this because the `AbstractBasicTranslationRotation3D` of Magnum does not allow setting the transformation matrix. `Eigen::Isometry3d` is just a matrix that allows storing isometric transformations (i.e., rigid body). How can I set the matrix of `AbstractBasicTranslationRotation3D`? Another thing would be to check if Dart has some helpers to get it in Quaternions or something..